src/main/java/com/tira/restaurants/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,55 @@
+package com.tira.restaurants.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.web.authentication.logout.HttpStatusReturningLogoutSuccessHandler;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    @Bean
+    public BCryptPasswordEncoder bCryptPasswordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+        		.csrf()
+        			.disable()
+                .authorizeRequests()
+                    .antMatchers("/register","/login").permitAll()
+                    .anyRequest().authenticated()
+                    .and()
+//                 .formLogin()
+//                 	.loginProcessingUrl("/login")
+//                 	.usernameParameter("email").passwordParameter("password")
+//                 .and()
+                 .logout().logoutSuccessHandler((new HttpStatusReturningLogoutSuccessHandler(HttpStatus.OK))).permitAll()
+                    .and()
+                 .rememberMe()
+                    .rememberMeParameter("rememberMe")
+                    .key("rememberMe");
+        
```

Comment:
I tried implementing rememberMe functionality using this HttpSecurity method rememberMe but it takes the rememberMe as a form parameter by default, so I was not sure how to handle the rememberMe parameter sent in request body as json.

Reply:
What about fetching the rememberMe field from the database?